coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -3759,39 +3765,190 @@ def _internal_op_tensor_inplace_fill(context, node):
 
 
 @register_torch_op
-def index_put(context, node):
+def select_scatter(context, node):
     inputs = _get_inputs(context, node, expected=4)
     x = inputs[0]
+    updates = inputs[1]
+    dim = inputs[2].val
+    index = inputs[3]
+
+    # mb.torch_tensor_assign handles multi-dim slicing
+    # so we need to create slice specifications for all other dimensions
+    begin = [0] * x.rank
+    begin[dim] = index
+    begin = mb.concat(values=begin, axis=0)
+    end = x.shape
+    stride = [1] * x.rank
+
+    begin_mask = [True] * x.rank
+    if index.val not in (0, -x.rank):
```

Comment:
This logic seems redundant for me.
Can't we just set `begin_mask[dim] = False` for all cases?
We are respecting the `begin[dim]` for all cases anyway.

Reply:
Very good point, I have a question about all our slice ops:
1. Why we have both `begin` and `begin_mask`?
2. Shouldn't `begin` be the single source of truth?
3. The only reasonable reason I can guess is "begin_mask True is more performant than begin 0", is this correct?

Reply:
Yeah but say `index.val == 0 and begin_mask=False`, the slicing is still starting at `0`.
So the `begin_mask[dim]` can always be `False`, regardless the value of `index`

Reply:
Discussed offline. Since mask is intended for flexibility of special cases (e.g. `step < 0` and `begin_mask = True` on dynamic shape), here we have no need to use mask at all: just use `begin` as single source of truth